script.js

```diff
@@ -1,12 +1,178 @@
+  episodesEl.setAttribute('style', 'display:inline');
+  let allEpisodesHTML = '';
+  arr.forEach(episode => allEpisodesHTML += getEpisodeCard(episode));
+  episodesEl.innerHTML = allEpisodesHTML;
+  searchResult.innerHTML = `Displaying ${arr.length}/${arr.length}`;
+  searchBar(arr);
+}
+
+function createEpisodeSelect(arr) {
+  selectEpisodesEl.innerHTML = '';
+  let selectAllEpisodes = document.createElement('option');
+  selectAllEpisodes.innerText = 'All Episodes';
+  selectAllEpisodes.value = 'selectAll';
+  selectEpisodesEl.appendChild(selectAllEpisodes);
+  arr.forEach(episode => {
+    let selectOption = document.createElement('option');
+    selectOption.innerText = `${getEpisodeTitle(episode)}`;
+    selectOption.value = episode.id;
+    selectEpisodesEl.appendChild(selectOption);
+  });
+}
+
+function displaySelectedShow(arr) {
+  createEpisodeSelect(arr);
+  displayEpisodes(arr);
+  searchBar(arr);
+  searchInput.value = '';
+}
+
+selectEpisodesEl.addEventListener('change', (e) => {
+  if (e.target.value == 'selectAll') { displayEpisodes(allEpisodes); }
+  for (let ep of allEpisodes) {
+    if (ep.id == selectEpisodesEl.value) {
+      episodesEl.innerHTML = getEpisodeCard(ep);
+      searchResult.innerHTML = `Displaying 1/${allEpisodes.length}`;
+    }
+  }
+})
+
+selectShowEl.addEventListener('change', (e) => {
+  if (e.target.value == 'allShows') {
+    displayShows(allTVShows);
+    searchBar(allTVShows);
+  };
+  for (let show of allTVShows) {
+    if (show.id == selectShowEl.value) { fetchShowAndDisplay(show.id); }
+  }
+})
+
+function createShowSelect(arr) {
+  selectShowEl.innerHTML = '';
+  let selectAllShows = document.createElement('option');
+  selectAllShows.innerText = 'All Shows';
+  selectAllShows.value = 'allShows';
+  selectShowEl.appendChild(selectAllShows);
+  arr.forEach(show => {
+    let selectOption = document.createElement('option');
+    selectOption.innerText = show.name;
+    selectOption.value = show.id;
+    selectShowEl.appendChild(selectOption);
+  })
+}
+
+function displayShows(arr) {
+  showsEl.setAttribute('style', 'display:inline');
+  episodesEl.setAttribute('style', 'display:none !important');
+  let allShowsHTML = '';
+  arr.forEach(show => allShowsHTML += getShowCard(show))
+  showsEl.innerHTML = allShowsHTML;
+  searchResult.innerHTML = `Displaying ${arr.length}/${arr.length}`;
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+function displayAllShows(arr) {
+  searchInput.value = '';
+  createShowSelect(arr);
+  displayShows(arr);
+  searchBar(arr);
+}
+
+
+function getAllGenres() {
+  let genres = [];
+  allTVShows.forEach(show => show.genres.forEach(gen => genres.push(gen)));
+  allGenres = [...new Set(genres)];
 }
 
-window.onload = setup;
+function searchBar(arr) {
+  if (arr[0].genres != null) {
+    searchInput.addEventListener('keyup', (input) => {
```

Comment:
Your search method causing your page too slow. I tried to find where is the problem, but I couldn't define it exactly. Sorry, but I think it maybe because of the filter, I am not sure.

Reply:
I realised now my search is blocking my page... maybe is better if I make 2 search bar and show just one of them

Reply:
I have improved the search function, let me know what you think and can do better